feat(marketing-analytics): enhance dynamic conversion goal

jabahamondes opened the pull request:
## Problem

- when saving the conversion goal, the table didn't update (because the query remain the same so the cache was preventing to see the actual result)
- sorting was only saved in the state, refreshing the page would clear the sorting

## Changes

- Added conversion goals in webAnalyticsLogic so the query is re-rendered based on the conversion goal changes
- Moved DynamicConversionGoal logic to the marketing logic (this will in the future allow me to add a configuration modal and also use the dynamic and local conversion goal actions there)
- Improved MarketingAnalyticsTable to allow saving sorting in the url
- Refactored logic to ensure unique naming for conversion goals and reset functionality.
- Added an icon to differentiate the dynamic conversion goal from the rest

## How did you test this code?
Locally
<img width="1236" height="491" alt="image" src="https://github.com/user-attachments/assets/7011b0a7-cb9a-49a2-aeb7-be23e0a7ec4a" />

Commit:
feat(marketing-analytics): enhance dynamic conversion goal handling and UI interactions

- Added support for multiple conversion goals in webAnalyticsLogic.
- Updated DynamicConversionGoalControls to manage local conversion goals more effectively.
- Improved MarketingAnalyticsTable to allow saving conversion goals directly from sortable columns.
- Refactored logic to ensure unique naming for conversion goals and reset functionality.

This update streamlines the user experience for managing conversion goals in the marketing analytics section.

Commit:
merge conflicts

Commit:
Updated the icon for conversion goals from IconStarFilled to IconPlus in MarketingAnalyticsTable

Commit:
handling null case for localConversionGoal

lricoy reviewed (approved):
Nit(naming): Took me a while to grasp the difference on those, I think a single state for the conversion goal with a status to differentiate would work nicely.

Maybe a `draftConversionGoal/activeConversionGoal` works as well.

Commit:
Merge remote-tracking branch 'origin/master' into feat/saving-into-url-and-fixing-state-bug

Commit:
refactor: rename dynamicConversionGoal to draftConversionGoal across schema and components

Updated the schema and related components to reflect the change from 'dynamicConversionGoal' to 'draftConversionGoal'. This includes adjustments in TypeScript interfaces, UI components, and backend query handling to ensure consistency and clarity in the codebase.

Commit:
fixing test
